src/main/scala/piconot/internal/ourDSL.scala

```diff
@@ -0,0 +1,132 @@
+        case East  => eastWall = Blocked
+        case West  => westWall = Blocked
+        case South => southWall = Blocked
+      }
+    }
+    OurSurroundings(northWall, eastWall, westWall, southWall)
+  }
+
+  // Similar to blocked, but checks if the relative description is open
+  def open(walls: MoveDirection*): OurSurroundings = {
+    var northWall, eastWall, westWall, southWall: RelativeDescription = Anything
+    walls.foreach { dir =>
+      dir match {
+        case North => northWall = Open
+        case East  => eastWall = Open
+        case West  => westWall = Open
+        case South => southWall = Open
+      }
+    }
+    OurSurroundings(northWall, eastWall, westWall, southWall)
+  }
+
+  // Casts a tuple to ShortRule
+  implicit def tupletoShortRule(tuple: (Surroundings, MoveDirection, State)): ShortRule = {
+    ShortRule(s = tuple._1, m = tuple._2, state = tuple._3)
+  }
+
+  // ShortRule expects a Surroundings, so we explicitly construct one from OurSurroundings.
+  // Making an alternate constructor for ShortRule that takes an OurSurroundings didn't work for some reason (even though I tried using the 'new' keyword).
+  implicit def tupletoShortRule2(tuple: (OurSurroundings, MoveDirection, State)): ShortRule = {
+    ShortRule(s = Surroundings(tuple._1.north, tuple._1.east, tuple._1.west, tuple._1.south), m = tuple._2, state = tuple._3)
+  }
+  
+  // Implicit cast from string to State
+  implicit def stringToState(state: String): State = State(state)
+  
+  // Default vals for natural language replacement of cardinal directions
+  val left, west, w = West
+  val up, north, n = North
+  val right, east, e = East
+  val down, south, s = South
+  val stay, stop = StayHere
+
+  // Testing purposes
+  val defaultSurrounding: Surroundings = Surroundings(Anything, Anything, Anything, Anything)
+  val defaultMove: MoveDirection = StayHere
+  val defaultState: State = State("Default")
+  val defaultRule: Rule = Rule(
+    defaultState,
+    defaultSurrounding,
+    defaultMove,
+    defaultState)
+}
+
+case class ShortRule(s: Surroundings, m: MoveDirection, state: State) {
+  def this(s2: OurSurroundings, m2: MoveDirection, state2: State) {
+    this(Surroundings(s2.north, s2.east, s2.west, s2.south), m2, state2)
+  }
+}
+
+//By creating our own class that mimics Surroundings, we can have binary operators.  
+case class OurSurroundings(north: RelativeDescription, east: RelativeDescription, west: RelativeDescription, south: RelativeDescription) {
+  def +(other: OurSurroundings): Surroundings = {
+    /*
+     * @return whichever RelativeDescription is more specific (ie not Anything), or Anything if neither is specific.
+     */
+    def pickDir(d1: RelativeDescription, d2: RelativeDescription): RelativeDescription = {
+      if (!d1.equals(Anything)) {
+        println("d1")
+        return d1
+      } else if (!d2.equals(Anything)) {
+        println("d2")
+        return d2
+      }
+      Anything
+    }
```

Comment:
Here is where you could implement error checking. Let me clarify: while your code won't break if a user defines a direction as both blocked() and open(), I do think if a user does this, it is probably unintentional. This is because _Anything_ is the default value that occurs whenever a user doesn't specify anything. Thus, I think if a user types the extra material it is likely a mistake and should throw an error.